### ISSUE TITLE
Use vectorcall for clang as well

### ISSUE THREAD

**Issue by logankaser** [nfrechette/rtm#145]
Clang supports https://clang.llvm.org/docs/AttributeReference.html#vectorcall
so, it could be used when compiling with clang instead in addition to just MSVC

**Comment by nfrechette** [comment]
Good idea, I hadn't noticed this. There seem to be subtle differences between the default calling convention with clang and vectorcall. Clang will pass up to 8 vectors by value in registers but it doesn't appear to return aggregates by value and I'm not sure if it passes aggregates by value, I'd have to double check again. On the other hand, vector supports up to 6 arguments by value but it handles aggregates better.

We have to measure in a real application what the impact is to make sure it yields a net win.

**Comment by logankaser** [comment]
It says on that page:
> Homogeneous vector aggregates of up to four elements are passed in sequential SSE registers if enough are available
But of course for this kind of thing, testing/benchmarking is definitely the way to go

**Comment by nfrechette** [comment]
I just took a look at this a few days ago and I'll bump the priority of this. On Windows, when Clang is used, the calling convention is the default one for Windows which means that all SIMD types are passed by `const&` which is quite terrible for performance and codegen. There, `vectorcall` makes a HUGE difference. However, on Linux and OS X, the default calling convention doesn't suffer the same fate and SIMD types are passed by value/register as current handled by RTM.

I'll thus enable `vectorcall` usage with Clang on Windows.
